perfkitbenchmarker/providers/ibmcloud/ibmcloud_network.py

```diff
@@ -0,0 +1,243 @@
+      'vpcid': self.vpcid,
+      'zone': self.zone,
+      'name': self.prefix + VPC_NAME + util.SUBNET_SUFFIX + \
+      str(subnet_index) + util.DELIMITER + self.zone,
+      'cidr': VPC_SUBNETS[subnet_index - 1]
+      })
+    logging.info('Creating subnet: %s', cmd.flags)
+    resp = cmd.CreateSubnet()
+    if resp:
+      self.subnet = resp.get('id')
+      IbmCloudVirtualMachine.validated_subnets += 1
+      logging.info('Created subnet: %s, zone %s', self.subnet, self.zone)
+    else:
+      raise errors.Error('IBM Cloud ERROR: Failed to create subnet')
+
+  def CreateFip(self, name: str, vmid: str):
+    """Creates a VNIC in a IBM Cloud VM instance."""
+    cmd = ibm.IbmAPICommand(self)
+    cmd.flags['instanceid'] = vmid
+    vnicid = cmd.InstanceGetPrimaryVnic()
+    cmd.flags['name'] = name
+    cmd.flags['target'] = vnicid
+    logging.info('Creating FIP for instanceid: %s', vmid)
+    resp = json.loads(cmd.InstanceFipCreate())
+    if resp:
+      logging.info('FIP create resp: %s', resp)
+      assert resp['address'] != None
+      return resp['address'], resp['id']
+    logging.error('FAILED to create FIP for instance %s', vmid)
+    return None, None
+
+  def DeleteFip(self, vmid: str, fip_address: str, fipid: str):
+    """Deletes fip in a IBM Cloud VM instance."""
+    logging.info('Deleting FIP, instanceid: %s, fip address: %s, fip id: %s',
+                 vmid, fip_address, fipid)
+    cmd = ibm.IbmAPICommand(self)
+    cmd.flags['fipid'] = fipid
+    cmd.InstanceFipDelete()
+
+  def _Exists(self):
+    """Returns true if the VPC exists."""
+    cmd = ibm.IbmAPICommand(self)
+    cmd.flags.update({
+      'prefix': self.prefix,
+      'zone': self.zone,
+      'items': 'vpcs'
+      })
+    self.vpcid = cmd.ListResources()
+    if self.vpcid:
+      return True
+    return False
+
+  def _Delete(self):
+    """Deletes the vpc and subnets."""
+    with self._lock:
+      if IbmCloudVirtualMachine.validated_subnets > 0:
+        cmd = ibm.IbmAPICommand(self)
+        cmd.flags['items'] = 'subnets'
+        cmd.flags['id'] = self.subnet
+        logging.info('Waiting to delete subnet: %s', self.subnet)
+        time.sleep(30)
+        cmd.DeleteResource()
+        IbmCloudVirtualMachine.validated_subnets -= 1
+
+    # different lock so all threads get a chance to delete its subnet first
+    with self._lock_vpc:
+      if self.vpcid not in IbmCloudVirtualMachine.validated_resources_set:
+        cmd = ibm.IbmAPICommand(self)
+        # check till all subnets are gone
+        time_to_end = time.time() + _DEFAULT_TIMEOUT
+        while IbmCloudVirtualMachine.validated_subnets > 0:
+          logging.info('Subnets not empty yet')
+          if time_to_end < time.time():
+            break
+          time.sleep(10)
+        cmd.flags['items'] = 'vpcs'
+        cmd.flags['id'] = self.vpcid
+        logging.info('Waiting to delete vpc')
+        time.sleep(10)
+        IbmCloudVirtualMachine.validated_resources_set.add(self.vpcid)
```

Comment:
I haven't looked at this code but it seems odd to need / want to do this.

Reply:
This is to take care of cleaning up the vpc when vms run in different zones.